lib/game.rb

```diff
@@ -0,0 +1,66 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2], [3, 4, 5],
+    [6, 7, 8], [0, 3, 6],
+    [1, 4, 7], [2, 5, 8],
+    [0, 4, 8], [6, 4, 2]
+]
+
+  def initialize(player_1 = Human.new("X"), player_2 = Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2 # calls #turn_count method from Board class, if it returns even, it's player_1 is current_player
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |win_combo| # iterates over each element, which are nested arrays, in the WIN_COMBINATIONS []
+      # .all? passes each element of win_combo to black, returns true if the block never returns false or nil
+      win_combo.all? { |win_play| board.cells[win_play] == "X" } || win_combo.all? { |win_play| board.cells[win_play] == "O" }
+    end
+  end
+
+  def draw?
+    !won? && full?
+  end
+
+  def full?
+    board.cells.all? { |token| token == "X" || token == "O" } # returns true if elements in cells [] have an "X" or "O"
+  end
+
+  def winner
+    if won?
+      win_combo = won?
```

Comment:
You can accomplish both of these steps in a single line of code.
```
if win_combo = won?
```
If `won?` returns false, the result of the assignment operation will also be false, so the interior of the if statement will not be executed. This pattern is sometimes referred to as 'if assignment.'